clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6914,6 +6925,8 @@ static void ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D,
   // Unknown attributes are automatically warned on. Target-specific attributes
   // which do not apply to the current target architecture are treated as
   // though they were unknown attributes.
+  if (isAttrValidForSYCLDevice(S.Context, AL)) {
+  } else
```

Comment:
This should be rewritten as:
if (!isAttrValidForSYCLDevice(S.Context, AL) &&
    ...

Reply:
Thanks.  I will change that back.

Reply:
Changed